### ISSUE TITLE
Vulkan compile fails on OS X

### ISSUE THREAD

**Issue by buzmeg** [gfx-rs/gfx#2796]
I just pulled the latest Vulkan SDK for OS X (1.1.106.0) from here:
https://vulkan.lunarg.com/sdk/home

and attempted to compile it and got the following error:

```
$ cargo run --bin quad --features=vulkan
   Compiling gfx-backend-vulkan v0.2.0 (/Users/andrewl/rust/gfx/src/backend/vulkan)
error[E0463]: can't find crate for `x11`
  --> src/backend/vulkan/src/lib.rs:22:1
   |
22 | extern crate x11;
   | ^^^^^^^^^^^^^^^^^ can't find crate
```

Thanks.

**Comment by grovesNL** [comment]
In general `--features metal` should be strongly preferred when using gfx-hal on macOS/iOS. gfx-hal also translates Vulkan to Metal, so neither [gfx-portability](https://github.com/gfx-rs/portability) nor MoltenVK should be necessary. Basically `--features vulkan` would mean it goes through a another mapping layer on macOS:

- If gfx-portability is used, then it means running gfx-hal on top of gfx-portability. gfx-portability is a Vulkan Portability implementation built on gfx-hal. So this would involve running gfx-hal on top of itself, instead of using gfx-hal's Metal backend directly.
- If MoltenVK is used (the default Vulkan Portability implementation shipped with the SDK), then it means running gfx-hal on top of MoltenVK which eventually still calls Metal, instead of using it directly in gfx-hal.

Although the performance would be reduced, there might still be some value in making `--feature vulkan` work on macOS/iOS to be able to use the Vulkan validation layers there for debugging. It could also be used as a temporary workaround for use with older macOS versions until we're able to fix the swapchain issue from #2563.

**Comment by buzmeg** [comment]
> It could also be used as a temporary workaround for use with older macOS versions until we're able to fix the swapchain issue from #2563.

This was my intent, but not simply limited in scope to that bug.  At this point, I view the continuing existence of the swapchain bug (and others) as indicative that the OS X Metal system has enough uniqueness to the API/OS interactions that the Metal feature for gfx is always going to be short of development resources given the complicated Venn diagram of skills required in the programmers who can actually solve these bugs. 

I was hoping to swap out a dependency which has a very small number of programmers (Metal GFX backend on OS X) with two *DRAMATICALLY* higher priority dependencies (Vulkan GFX backend which gets Linux and Windows attention) and MoltenVK on OS X (which has its own momentum now that Valve is shoving it forward).

While this may or may not make the situation better in the short term, it would at least punt the Metal-specific debugging into a team whose actual paid job is to focus on OS X and Metal.

**Comment by grovesNL** [comment]
For what it's worth, we have a lot of users and projects running the Metal backend on macOS 10.13+ successfully – the issues mentioned only apply to macOS 10.12 and below. We still need to fix those issues, and we should definitely improve testing coverage on macOS 10.12 and below. But I don't think recommending MoltenVK over our own Metal backend instead is the right solution.

As mentioned above, I agree it's fine to provide minimal support for running macOS with Vulkan Portability, but it shouldn't be the primary solution we should recommend (primarily because gfx-hal itself is part of a Vulkan Portability implementation).

**Comment by aleksijuvani** [comment]
I believe this can be closed as of #2828?
